File: HRMS/HRMS.Web/Modules/Recruitment/Interviews/InterviewsColumns.cs
using Serenity.ComponentModel;

namespace HRMS.Recruitment.Columns;

[ColumnsScript("Recruitment.Interviews")]
[BasedOnRow(typeof(InterviewsRow), CheckNames = true)]
public class InterviewsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int InterviewId { get; set; }
    [EditLink]
    public string CandidateName { get; set; }
    public string InterviewerName { get; set; }
    public DateTime InterviewDate { get; set; }
    public int Round { get; set; }
    public int Rating { get; set; }
    public bool IsCompleted { get; set; }
    public DateTime? CompletedOn { get; set; }
    public string Comments { get; set; }
}

File: HRMS/HRMS.Web/Modules/ServerTypes/Recruitment/InterviewsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { InterviewRound } from "./InterviewRound";
import { InterviewsRow } from "./InterviewsRow";

export interface InterviewsColumns {
    InterviewId: Column<InterviewsRow>;
    CandidateName: Column<InterviewsRow>;
    InterviewerName: Column<InterviewsRow>;
    InterviewDate: Column<InterviewsRow>;
    Round: Column<InterviewsRow>;
    Rating: Column<InterviewsRow>;
    IsCompleted: Column<InterviewsRow>;
    CompletedOn: Column<InterviewsRow>;
    Comments: Column<InterviewsRow>;
}

export class InterviewsColumns extends ColumnsBase<InterviewsRow> {
    static readonly columnsKey = 'Recruitment.Interviews';
    static readonly Fields = fieldsProxy<InterviewsColumns>();
}

[InterviewRound]; // referenced types